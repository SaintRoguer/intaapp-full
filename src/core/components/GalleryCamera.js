import {Container, Content, Spinner} from 'native-base';
import React, {Component} from 'react';
import {StyleSheet} from 'react-native';
import {mainThemeColor, galleryCameraButtons} from '../../configuration';
import {
  HomeCard,
  withImagePicker,
  withImageProcessing,
  withImageHandler,
} from '../../shared';
import {ImageModel} from '../../shared/models/ImageModel.js';

class GalleryCamera extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
    };
  }

  render() {
    return (
      <Container>
        <Content contentContainerStyle={styles.container}>
          {this.state.loading ? (
            <Spinner color={mainThemeColor(1)} />
          ) : (
            <>{this.getGalleryCameraButtons()}</>
          )}
        </Content>
      </Container>
    );
  }

  getGalleryCameraButtons() {
    return galleryCameraButtons.map((button, index) => (
      <HomeCard
        key={index}
        onPress={this.launch(button.type)}
        icon={button.icon}
        text={button.text}
      />
    ));
  }

  launch = picker => async () => {
    try {
      this.setState({loading: true});
      const imageResponse = await this.props.imageHandler.pickImage()(picker);
      this.props.navigation.navigate('Imagen', imageResponse);
    } finally {
      this.setState({loading: false});
    }
  };
}

export default withImageHandler(GalleryCamera);
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
