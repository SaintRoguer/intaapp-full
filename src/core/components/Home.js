import {Container, Content, Spinner} from 'native-base';

import React, {Component} from 'react';
import {StyleSheet, Alert} from 'react-native';
import {mainThemeColor, homeButtons} from '../../configuration';
import {HomeCard, ImagePickerService, ImageProcessor} from '../../shared';

export class Home extends Component {
  picker;
  imageProcessor;

  constructor(props) {
    super(props);
    this.picker = ImagePickerService.getInstance();
    this.imageProcessor = ImageProcessor.getInstance();
    this.state = {
      loading: false,
    };
  }

  render() {
    return (
      <Container>
        <Content contentContainerStyle={styles.container}>
          {this.state.loading ? (
            <Spinner color={mainThemeColor(1)} />
          ) : (
            <>{this.getHomeButtons()}</>
          )}
        </Content>
      </Container>
    );
  }

  getHomeButtons() {
    return homeButtons.map((button, index) => (
      <HomeCard
        key={index}
        onPress={this.launch(button.type)}
        icon={button.icon}
        text={button.text}
      />
    ));
  }

  launch = picker => async () => {
    this.props.navigation.navigate('GalleryCamera', {});
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    // backgroundColor: '#20d2bb',
  },
});
