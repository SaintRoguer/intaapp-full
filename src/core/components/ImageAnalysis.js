//@ts-check
import React, {useState} from 'react';
import {StyleSheet, View} from 'react-native';
import {SaveImage, ImageView, Tabs, ImageNotes} from '../../shared';
import {withImageProcessing} from '../../shared';

function ImageAnalysis({route: {params}, imageProcessor}) {
  const onSave = params.onSave;
  const [originalImage, setOriginalImage] = useState(params.originalImage);
  const [processedImage, setProcessedImage] = useState(params.processedImage);
  const [percentages, setPercentages] = useState(params.percentages);
  const [note, setNote] = useState('');

  const updateOriginalImage = async newUri => {
    let {
      percentageGreen,
      percentageYellow,
      percentageNaked,
      img,
    } = await imageProcessor.processImage('file://' + newUri);
    setPercentages({
      percentageGreen,
      percentageYellow,
      percentageNaked,
    });

    setOriginalImage(prev => {
      const newOriginal = prev;
      newOriginal.setUri('file://' + newUri);
      return newOriginal.clone();
    });
    setProcessedImage(prev => {
      const newProcessed = prev;
      newProcessed.setData(img);
      return newProcessed.clone();
    });
  };

  const saveImage = async () => {
    return await onSave(percentages, originalImage.getSource(), note);
  };

  const getImageView = () => (
    <ImageView
      shouldRotate={params.shouldRotate}
      updateOriginalImage={updateOriginalImage}
      originalImage={originalImage}
      staticOriginal={params.originalImage.clone()}
      processedImage={processedImage}
      percentages={percentages}
    />
  );

  return (
    <View style={styles.container}>
      {onSave ? (
        <Tabs
          FirstScreen={getImageView}
          SecondScreen={() => <ImageNotes value={note} updateNote={setNote} />}
          secondTitle="Notas"
          firstTitle="AnÃ¡lisis"
        />
      ) : (
        getImageView()
      )}
      {onSave ? <SaveImage onSave={saveImage} /> : null}
    </View>
  );
}
export default withImageProcessing(ImageAnalysis);

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  information: {
    flex: 1,
  },
  switchContainer: {
    width: '100%',
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  switch: {
    width: '80%',
    zIndex: 99,
  },
});
