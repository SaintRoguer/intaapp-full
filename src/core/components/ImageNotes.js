import React from 'react';
import {View} from 'react-native';

export function ImageNotes() {
  return <View />;
}
