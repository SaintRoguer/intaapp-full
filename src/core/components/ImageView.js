import React, {Component} from 'react';
import {StyleSheet, View} from 'react-native';
import SwitchSelector from 'react-native-switch-selector';
import {options, switchConfig} from '../../configuration';
import {
  ImageProcessor,
  ImageWithAdjustment,
  FirebaseService,
  CloudinaryService,
} from '../../shared';
import {Percentages} from './Percentages';

export class ImageView extends Component {
  imageProcessor;
  originalImage;
  cloudinaryService;

  constructor(props) {
    super(props);
    this.state = {
      showOriginal: false,
      ...this.props.route.params,
    };
    this.originalImage = this.state.originalImage.clone();
    this.imageProcessor = ImageProcessor.getInstance();
    this.cloudinaryService = CloudinaryService.getInstance();
    this.firebaseService = FirebaseService.getInstance();
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.switchContainer}>
          <SwitchSelector
            style={styles.switch}
            options={options}
            onPress={value =>
              this.setState({
                showOriginal: value,
              })
            }
            {...switchConfig}
          />
        </View>
        <ImageWithAdjustment
          onImageAdjusted={this.updateOriginalImage}
          imageToEdit={this.originalImage}
          shouldRotate={this.props.route.params.shouldRotate}
          imageToShow={this.getImage()}
        />
        <Percentages
          percentageGreen={this.state.percentageGreen}
          percentageYellow={this.state.percentageYellow}
          percentageNaked={this.state.percentageNaked}
        />
      </View>
    );
  }

  updateOriginalImage = async newUri => {
    let nativeReponse = await this.imageProcessor.processImage(
      'file://' + newUri,
    );

    this.setState(async prevState => {
      let newState = {
        ...prevState,
      };
      newState.originalImage.setUri('file://' + newUri);
      newState.processedImage.setData(nativeReponse.img);
      newState.percentageGreen = nativeReponse.percentageGreen;
      newState.percentageYellow = nativeReponse.percentageYellow;
      newState.percentageNaked = nativeReponse.percentageNaked;

      // // const uri = await this.cloudinaryService.uploadPhoto(
      // //   `data:image/png;base64,${newState.processedImage.getData()}`,
      // // );
      // const uri = await this.cloudinaryService.uploadPhoto(
      //   newState.originalImage.getUri(),
      // );
      // if (uri != null) {
      //   this.firebaseService.uploadPhoto(uri);
      // }
      return newState;
    });
  };

  getImage() {
    let sourceOfImage = this.state.showOriginal
      ? this.state.originalImage
      : this.state.processedImage;
    return sourceOfImage;
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  information: {
    flex: 1,
  },
  switchContainer: {
    width: '100%',
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  switch: {
    width: '80%',
    zIndex: 99,
  },
});
