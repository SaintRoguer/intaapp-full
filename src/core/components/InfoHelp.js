//@ts-check
import React, {useState, useEffect} from 'react';
import {StyleSheet, View, Text} from 'react-native';
import {HeaderBackButton} from '@react-navigation/stack';

function InfoHelp({navigation}) {
  useEffect(() => {
    navigation.setOptions({
      headerLeft: () => (
        <HeaderBackButton
          tintColor="white"
          //   onPress={() => {
          //     navigation.navigate('Main', {
          //       showInfo: false,
          //     });
          //   }}
        />
      ),
    });
  }, []);

  return (
    <View style={styles.container}>
      <Text>INfo</Text>
    </View>
  );
}
export default InfoHelp;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
