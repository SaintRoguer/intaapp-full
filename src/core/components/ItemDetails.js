//@ts-check
import React, {useState, useEffect} from 'react';
import {Text, StyleSheet, View} from 'react-native';
import {
  withAlertService,
  withFirebase,
  Tabs,
  uniqueId,
  GridWithNewButton,
  ImagesTaken,
  BottomRightButton,
  Info,
  withImageHandler,
} from '../../shared/';

function ItemDetails({
  firebaseService: firebase,
  alertService: alerts,
  navigation,
  imageHandler,
}) {
  const [images, setImages] = useState([]);
  const [pasturas, setPasturas] = useState([]);
  const [itemDetail, setItemDetail] = useState();
  const {item} = navigation.state.params;
  useEffect(() => {
    firebase
      .getDataFromInnerId('lotesDetails')(item.id)
      .then(item => {
        setItemDetail(item);
        setImages(item.images);
        setPasturas(item.pasturas);
      });
  }, [item.id]);
  const handleImage = imageHandler.pickImage({
    collectionName: 'lotesDetails',
    itemId: item.id,
  });
  const noop = () => {};
  return (
    <View style={styles.detailsContainer}>
      <Info item={itemDetail} />
      <Tabs
        firstTitle="Pasturas"
        secondTitle="Imagenes"
        FirstScreen={() => (
          <GridWithNewButton
            title=""
            data={pasturas}
            onEntryClick={noop}
            onNewClick={noop}
            onDeleteEntry={noop}
          />
        )}
        SecondScreen={() => <ImagesTaken images={images} loteId={item.id} />}
      />
      <BottomRightButton
        withBackground={true}
        buttons={[
          {
            name: 'upload',
            type: 'FontAwesome5',
            onPress: async () => {
              const imageResponse = await handleImage('Gallery');
              navigation.navigate('Imagen', imageResponse);
            },
          },
          {
            name: 'camera-retro',
            type: 'FontAwesome5',
            onPress: async () => {
              const imageResponse = await handleImage('Camera');
              navigation.navigate('Imagen', imageResponse);
            },
          },
        ]}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  detailsContainer: {
    height: '100%',
  },
  description: {
    fontSize: 21,
    textAlign: 'center',
  },
});

export default withImageHandler(withAlertService(withFirebase(ItemDetails)));
