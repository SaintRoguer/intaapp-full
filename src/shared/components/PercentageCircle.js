import React from 'react';
import {Dimensions, Text, View, StyleSheet} from 'react-native';
import ProgressCircle from 'react-native-progress/Circle';

export function Percentage(props) {
  const formatText = progress => () => {
    if (props.hidePercentage) return '-';
    if (progress < 3) {
      return '< 3%';
    }
    return `${progress}%`;
  };

  return (
    <View style={styles.circle}>
      <Text style={styles.title}>{props.title}</Text>
      <ProgressCircle
        thickness={7}
        color={props.color}
        progress={props.percentage / 100}
        size={props.size || Dimensions.get('window').width * 0.2}
        showsText={true}
        formatText={formatText(props.percentage)}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  circle: {
    alignItems: 'center',
  },
  title: {
    fontSize: 16,
    color: 'white',
    marginBottom: 10,
  },
});
