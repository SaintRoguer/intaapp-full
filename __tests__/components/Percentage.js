/**
 * @format
 */
import 'react-native';
import React from 'react';
import {Percentage} from '../../src/shared/components/PercentageCircle';

import renderer from 'react-test-renderer';
import {ProgressCircle} from 'react-native-progress/Circle';

describe('Percentage component', () => {
  test('Renders correctly', () => {
    const component = renderer.create(<Percentage />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Renders with props', () => {
    const component = renderer.create(
      <Percentage color="green" percentage={66} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Has correct color', () => {
    const component = renderer.create(
      <Percentage color="green" percentage={66} />,
    );
    let root = component.root;
    const percentageCircle = root.findByType(ProgressCircle);
    expect(percentageCircle.props.color).toEqual('green');
  });
  test('Has correct value', () => {
    const component = renderer.create(
      <Percentage color="green" percentage={66} />,
    );
    let root = component.root;
    const percentageCircle = root.findByType(ProgressCircle);
    expect(percentageCircle.props.progress._value).toBeCloseTo(0.66, 5);
  });
});
