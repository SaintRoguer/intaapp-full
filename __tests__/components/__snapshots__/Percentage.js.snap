// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Percentage component Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  />
  <View
    indeterminateAnimationDuration={1000}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <ARTSurfaceView
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          },
          Object {
            "height": 150,
            "width": 150,
          },
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            75,
            4.5,
            2,
            NaN,
            NaN,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#007aff"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={7}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            75,
            0.5,
            4,
            75,
            75,
            74.5,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            75,
            75,
            74.5,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#007aff"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTSurfaceView>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 67,
          "height": 134,
          "justifyContent": "center",
          "left": 8,
          "position": "absolute",
          "top": 8,
          "width": 134,
        }
      }
    >
      <Text
        allowFontScaling={true}
        style={
          Array [
            Object {
              "color": "rgba(0, 122, 255, 1)",
              "fontSize": 29.77777777777778,
              "fontWeight": "300",
            },
            undefined,
          ]
        }
      >
        undefined%
      </Text>
    </View>
  </View>
</View>
`;

exports[`Percentage component Renders with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  />
  <View
    indeterminateAnimationDuration={1000}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <ARTSurfaceView
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          },
          Object {
            "height": 150,
            "width": 150,
          },
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            75,
            4.5,
            3,
            108.84208239100113,
            4.5,
            137.90987728962835,
            28.546971250916165,
            144.25125117637253,
            61.78961732170642,
            3,
            150.59262506311674,
            95.03226339249669,
            132.41835340491548,
            128.09114081301644,
            100.95278096426978,
            140.54924225512173,
            3,
            69.48720852362408,
            153.00734369722704,
            33.60837579509318,
            141.3496042658815,
            15.47488125210792,
            112.77578904601924,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#008000"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={7}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            75,
            0.5,
            4,
            75,
            75,
            74.5,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            75,
            75,
            74.5,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#008000"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTSurfaceView>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 67,
          "height": 134,
          "justifyContent": "center",
          "left": 8,
          "position": "absolute",
          "top": 8,
          "width": 134,
        }
      }
    >
      <Text
        allowFontScaling={true}
        style={
          Array [
            Object {
              "color": "green",
              "fontSize": 29.77777777777778,
              "fontWeight": "300",
            },
            undefined,
          ]
        }
      >
        66%
      </Text>
    </View>
  </View>
</View>
`;
